src/cse-machine/utils.ts

```diff
@@ -312,6 +312,26 @@ export const createBlockEnvironment = (
   }
 }
 
+export const createProgramEnvironment = (
+  context: Context,
+  isPrelude: boolean,
+  head: Frame = {}
+): Environment => {
+  return isPrelude
+    ? {
+        name: 'prelude',
+        tail: currentEnvironment(context),
+        head,
+        id: uniqueId()
+      }
+    : {
+        name: 'programEnvironment',
+        tail: currentEnvironment(context),
+        head,
+        id: uniqueId()
+      }
+}
+
```

Comment:
Can we just combine the two and only change the name property to prevent duplication (since only the name property differs)?


```suggestion
  return {
        name: isPrelude ? 'prelude' : 'programEnvironment',
        tail: currentEnvironment(context),
        head,
        id: uniqueId()
      }
}

```

Reply:
Yes that would be better.